presto-main/src/main/java/com/facebook/presto/operator/HashAggregationOperator.java

```diff
@@ -563,7 +598,18 @@ private void initializeAggregationBuilderIfNeeded()
             return;
         }
 
-        if (step.isOutputPartial() || !spillEnabled) {
+        boolean partialAggregationDisabled = partialAggregationController
```

Comment:
Can you think of any runtime or operator level stats to expose to count say the number of times that partial agg gets disabled ?